### PR TITLE
[ACS-5650] viewer action files e2e migration

### PR DESCRIPTION
**Pull request by akashrathod28**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe: protractor test migration to playwright

**What is the current behaviour?** (You can also link to an open issue here)
protractor e2e

**What is the new behaviour?**
migration to playwright e2e

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**: https://alfresco.atlassian.net/browse/ACS-5650

### CODE REVIEW

File: e2e/playwright/viewer/src/tests/viewer-action.spec.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  const randomDocxDelete = `${TEST_FILES.DOCX.name}-${Utils.random()}`;
+  const fileForEditOffline = `playwright-file1-${Utils.random()}.docx`;
+  const fileForCancelEditing = `playwright-file2-${Utils.random()}.docx`;
+  let folderId: string;
+  let fileDocxShareId: string;
+
+  test.beforeAll(async ({ fileAction, shareAction }) => {
+    await apiClientFactory.setUpAcaBackend('hruser');
+    const node = await apiClientFactory.nodes.createNode('-my-', { name: randomFolderName, nodeType: 'cm:folder', relativePath: '/' });
+    folderId = await node.entry.id;
```
Comment (MichalKinas) [inline e2e/playwright/viewer/src/tests/viewer-action.spec.ts:44]: You don't need `await` here, Promise will be resolved one line above

File: e2e/playwright/viewer/src/tests/viewer-action.spec.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    await fileAction.uploadFile(TEST_FILES.DOCX.path, fileForCancelEditing, folderId);
+    await fileAction.uploadFile(TEST_FILES.DOCX.path, randomDocxName, folderId);
+    await fileAction.uploadFile(TEST_FILES.DOCX.path, randomDocxDelete, folderId);
+    const fileDocShare = await fileAction.uploadFile(TEST_FILES.DOCX.path, randomDocxNameShare, folderId);
+    fileDocxShareId = await fileDocShare.entry.id;
```
Comment (MichalKinas) [inline e2e/playwright/viewer/src/tests/viewer-action.spec.ts:49]: Same here, `await` not needed

File: e2e/playwright/viewer/src/tests/viewer-action.spec.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    await fileAction.uploadFile(TEST_FILES.DOCX.path, randomDocxNameFavorite, folderId);
+    await shareAction.shareFileById(fileDocxShareId);
+    await fileAction.uploadFile(TEST_FILES.DOCX.path, fileForEditOffline, folderId);
+  });
+
+  test.beforeEach(async ({ personalFiles }) => {
+    const gotoNodeURL = `#/personal-files/${folderId}`;
```
Comment (MichalKinas) [inline e2e/playwright/viewer/src/tests/viewer-action.spec.ts:56]: You don't need to create this variable, you can just pass the url directly into `navigate`

File: projects/aca-playwright-shared/src/api/shared-links-api.ts [hunk 4]
```diff
@@ -52,4 +51,22 @@ export class SharedLinksApi extends ApiClientFactory {
       return null;
     }
   }
+
+  private async getFavorites(userName: string) {
```
Comment (MichalKinas) [inline projects/aca-playwright-shared/src/api/shared-links-api.ts:55]: Please add return type to non `void` methods

File: projects/aca-playwright-shared/src/api/shared-links-api.ts [hunk 5]
```diff
@@ -52,4 +51,22 @@ export class SharedLinksApi extends ApiClientFactory {
...
+    try {
+      return await this.apiService.favorites.listFavorites(userName);
+    } catch (error) {
+      logger.error(`\n--- ${error} error :`);
+      return null;
+    }
+  }
+
+  async isFavorite(nodeId: string, userName: string) {
```
Comment (MichalKinas) [inline projects/aca-playwright-shared/src/api/shared-links-api.ts:64]: Please add return type to non `void` methods

File: projects/aca-playwright-shared/src/api/shared-links-api.ts [hunk 6]
```diff
@@ -52,4 +51,22 @@ export class SharedLinksApi extends ApiClientFactory {
       return null;
     }
   }
+
+  private async getFavorites(userName: string) {
+    try {
+      return await this.apiService.favorites.listFavorites(userName);
+    } catch (error) {
+      logger.error(`\n--- ${error} error :`);
```
Comment (MichalKinas) [inline projects/aca-playwright-shared/src/api/shared-links-api.ts:59]: Could you maybe add some indicator of where error was thrown into the error message? Something like `Error while fetching favourites ${error} :`

File: projects/aca-playwright-shared/src/api/shared-links-api.ts [hunk 7]
```diff
@@ -52,4 +51,22 @@ export class SharedLinksApi extends ApiClientFactory {
...
+      return null;
+    }
+  }
+
+  async isFavorite(nodeId: string, userName: string) {
+    try {
+      return JSON.stringify((await this.getFavorites(userName)).list.entries).includes(nodeId);
+    } catch (error) {
+      logger.error(`\n--- ${error} error :`);
```
Comment (MichalKinas) [inline projects/aca-playwright-shared/src/api/shared-links-api.ts:68]: Same here with error message

File: e2e/playwright/viewer/src/tests/viewer-action.spec.ts (hunk 1 above)
Comment (akashrathod28) [reply e2e/playwright/viewer/src/tests/viewer-action.spec.ts]: removed await in 6022502476dfc693f8f53553773adb77403b3133

File: e2e/playwright/viewer/src/tests/viewer-action.spec.ts (hunk 2 above)
Comment (akashrathod28) [reply e2e/playwright/viewer/src/tests/viewer-action.spec.ts]: removed, 6022502476dfc693f8f53553773adb77403b3133

File: projects/aca-playwright-shared/src/api/shared-links-api.ts (hunk 6 above)
Comment (akashrathod28) [reply projects/aca-playwright-shared/src/api/shared-links-api.ts]: added error description.

**Review by MichalKinas — APPROVED** [review]

File: e2e/playwright/viewer/src/tests/viewer-action.spec.ts [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    await favoritePage.viewerDialog.favoriteMenuButton.waitFor({ state: 'attached', timeout: timeouts.normal });
+    await sharedPage.viewerDialog.favoriteMenuButton.click();
+    await favoritePage.viewerDialog.favoriteMenuButton.waitFor({ state: 'detached', timeout: timeouts.normal });
+
+    await sharedPage.acaHeader.clickViewerMoreActions();
+    expect(await sharedPage.viewerDialog.removeFavoriteMenuButton.isVisible(), 'Item should be remove favorite').toBe(true);
+    await sharedPage.page.keyboard.press('Escape');
+    await favoritePage.navigate({ waitUntil: 'domcontentloaded' });
+    expect(await favoritePage.dataTable.isItemPresent(randomDocxNameShare), 'Item is not present in Favorites list').toBe(true);
+    expect(await shareAction.isFavorite(randomDocxNameShare, 'hruser'), 'Item is not favorite').toBe(true);
```
Comment (azakrzewski-hy) [inline e2e/playwright/viewer/src/tests/viewer-action.spec.ts:149]: Should we do API validation in UI test?

File: e2e/playwright/viewer/src/tests/viewer-action.spec.ts [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+    await personalFiles.acaHeader.clickViewerMoreActions();
+    await personalFiles.viewerDialog.deleteMenuButton.click();
+    await personalFiles.snackBar.getByMessageLocator(randomDocxDelete).waitFor({ state: 'attached' });
+    const deleteName = await personalFiles.snackBar.getByMessageLocator(randomDocxDelete).innerText();
+    expect(deleteName).toContain(`${randomDocxDelete} deleted`);
+    await personalFiles.dataTable.getCellLinkByName(randomDocxName).waitFor({ state: 'attached' });
+    expect(await personalFiles.dataTable.getCellLinkByName(randomDocxDelete).isVisible(), 'file should not visible').toBe(false);
+    await trashPage.navigate({ waitUntil: 'domcontentloaded' });
+    await trashPage.dataTable.goThroughPagesLookingForRowWithName(randomDocxDelete);
+    expect(await trashPage.dataTable.isItemPresent(randomDocxDelete), 'Item should be present in Trash').toBe(true);
```
[the remark this answers is not in the document]
Comment (akashrathod28) [reply e2e/playwright/viewer/src/tests/viewer-action.spec.ts:86]: I will take care of this, currently this test is excluded as it is  taking more the 45sec to execute and and I will reenable it one ES changes merge.

File: e2e/playwright/viewer/src/tests/viewer-action.spec.ts (hunk 8 above)
Comment (akashrathod28) [reply e2e/playwright/viewer/src/tests/viewer-action.spec.ts:149]: Here, Its been done in protractor test, but we can remove this extra validation as not needed.

**Review by azakrzewski-hy — APPROVED** [review]

**Review by DenysVuika — APPROVED** [review]

**Pull request merged**